### PR TITLE
Add Skopeo task

### PR DESCRIPTION
**Pull request by Divyansh42**
# Changes

This is a part of documenting and providing `Skopeo` task, which can mirror a given set of images from a specified container image registry to another container image registry.

Signed-off-by: Divyansh42 <diagrawa@redhat.com>

authors: @divyansh42 , @vinamra28 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Yaml file complies with [yamllint](https://github.com/adrienverge/yamllint) rules.

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by vinamra28** [comment]
Used `command` and `args` instead of `script` so as to hide the credentials from the logs

File: skopeo/test/dest-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
```
Comment (hrishin) [inline skopeo/test/dest-secret.yaml:1]: Instead of passing the creds, could use such approach

Comment (vinamra28) [reply skopeo/test/dest-secret.yaml:1]: @hrishin which approach :sweat_smile: ?

Comment (hrishin) [reply skopeo/test/dest-secret.yaml:1]: lol... sorry I mean like https://github.com/tektoncd/catalog/blob/v1beta1/buildah/tests/pre-apply-task-hook.sh @vinamra28

Comment (vdemeester) [reply skopeo/test/dest-secret.yaml:1]: aka using the local registry :angel:

Comment (Divyansh42) [reply skopeo/test/dest-secret.yaml:1]: Okay, will update with that :slightly_smiling_face:

**Comment by Divyansh42** [comment]
@vdemeester can you please review this or merge the PR, if it looks good to you :sweat_smile:

**Comment by savitaashture** [comment]
@Divyansh42 
Overall looks good to me 
Thank you for the PR

Could you please add a test example for pipeline with different task pointing to different registry

**Comment by Divyansh42** [comment]
> @Divyansh42
> Overall looks good to me
> Thank you for the PR
> 
> Could you please add a test example for pipeline with different task pointing to different registry

Okay, will add.

Thanks for the review :)

**Comment by Divyansh42** [comment]
@bobcatfish  @dlorenc Can you please review this PR?

This is a Tekton task that will help to copy set of images from a specified container image registry to another container image registry. This can be done by following the steps given below:

- Provide image URL as the `params` the task.
- Provide the required credentials for the image registries, in the form of `secrets`.
- In case of `multiple` destination and source images,  `configMap` consisting of images separated by space which needs to be copied from particular source to particular destination should be mounted as a `workspace`.

**Comment by vinamra28** [comment]
@bobcatfish @dlorenc just bumping again in case you have missed this out 😅😅

File: skopeo/README.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+- **srcImageURL**: The URL of the image to be copied to the `destination` registry.
+- **destImageURL**: The URL of the image where the image from `source` should be copied to.
+
+## Resources
+
+* `Secret` to provide the credentials of the source and destination registry where the image needs to be copied from and to.
+* `ConfigMap` to provide support for copying multiple images, this contains file `url.txt` which stores images registry URL's.
+* `Workspace` to mount file containing multiple source and destination images registries URL, which is mounted as configMap.
+
+[This](../skopeo/example/quay-secret.yaml) example can help to use secrets for providing credentials.
```
Comment (vdemeester) [inline skopeo/README.md:58]: I don't think this should be `Resources` as usually we refer to `Resources` for the PipelineResource. 
You should have a `## Workspace` heading and one for configmap and secret most likely

Comment (Divyansh42) [reply skopeo/README.md:58]: Okay, will update this. Thanks for the review :)

**Comment by Divyansh42** [comment]
@vdemeester, I have done the required changes :slightly_smiling_face:

**Pull request merged**